### DISCUSSION TITLE
Change the Footer per Chapter

### DISCUSSION THREAD

**Discussion by thekief** [typst/typst#3656] [General]
I am working on a pandoc template that needs to support preambles with roughly following structure:

- preamble
- table of contents
- content

I would like the preamble to have a roman lettering in the footer, the table of contents to have no footer, and the content some other footer layout. is there any workaround to do this? If possible I would like to do this directly in Typst rather than relying on some sort of external pre-processing.

**Comment by ludwig-austermann** [comment]
Since I suppose that you want the table of contents on a separate page (otherwise, the question is ambiguous), you can simply do something like:

```typst
#set page(numbering: "I")        // sets the following pages's page number style to roman

= Preamble
#lorem(900)

#set page(numbering: none)       // removes the pagenumbering from here on

#outline()

#counter(page).update(0)         // updates the page counter to again start from 1
#set page(numbering: "1")        // reactivates the page numbering with arabic style

= Content
== First Subcontent
#lorem(630)

== Second Subcontent
#lorem(1650)

== Third Subcontent
#lorem(700)
```

The `page` function currently automatically starts a new page when called, but you can combine it with a `pagebreak` for future compatibility.

**Reply by thekief** [reply]
My issue is that the focus is on the term `pandoc template`. When using pandoc, the template can make use of certain variables and for the textual part it is mainly `$body$`. After running pandoc, `$body$` is replaced with the document's content. There are no variables that allow the inclusion of a separate preamble or the like.

A template like would look like this:

```typst

#set page(numbering: none)       // removes the pagenumbering from here on

#outline()

#counter(page).update(0)         // updates the page counter to again start from 1
#set page(numbering: "1")        // reactivates the page numbering with arabic style

$body$
```

Resulting in something likes this after processing:

```typst

#set page(numbering: none)       // removes the pagenumbering from here on

#outline()

#counter(page).update(0)         // updates the page counter to again start from 1
#set page(numbering: "1")        // reactivates the page numbering with arabic style

= Preamble
#lorem(900)

= Content
== First Subcontent
#lorem(630)

== Second Subcontent
#lorem(1650)

== Third Subcontent
#lorem(700)
```

Thus, the preamble is after the ToC when automatically generated. With some pre-processing I could definitely change that but I was wondering if there is a Typst only solution.

**Comment by thekief** [comment]
Not quite beautiful but I added the following snippet:

```typst
#let add_preamble()= {
  set heading(numbering: none)

  let data = read("preamble.typ")

  if data.len() != 0 {
    set page(numbering: "I")
    counter(page).update(1)
    eval(data, mode: "markup")
  }
}
```
A major drawback is the needed existence of `preamble.typ` but that's my smallest sorrow in this case.
